### ISSUE TITLE
`unnecessary_find_map` should see through `then_some`

### ISSUE THREAD

**Issue by Swatinem** [rust-lang/rust-clippy#11260]
### Summary

It would be nice if Clippy could also warn me about using `then_some` in return of a `find_map` closure, like in this example:

```
std::iter::once(1).find_map(|n| (n > 1).then_some(n));
```

### Lint Name

unnecessary_find_map

### Reproducer

I tried this code:

```rust
    // warns
    std::iter::once(1).find_map(|n| if n > 1 { Some(n) } else { None });
    // does not
    std::iter::once(1).find_map(|n| (n > 1).then_some(n));
    // better solution
    std::iter::once(1).find(|n| *n > 1);
```

I expected to see this happen:

The `unnecessary_find_map` lint should flag the second example as well

Instead, this happened:

No warning is emitted

**Comment by Swatinem** [comment]
Wow, 8 hours from filing to resolution, solid work @y21 🥇
